### PR TITLE
[Hypershift] Filter regions where HostedCP is avalaible

### PR DESCRIPTION
**Pull request by andreadecorte**
Add a check at cluster creation to ensure region can host an HostedCP Add a filter in region list command for HostedCP enabled regions

Fixes [SDA-6899](https://issues.redhat.com//browse/SDA-6899)

### ORG HostedCP enabled 
rosa list regions --hosted-cp
ID           NAME               MULTI-AZ SUPPORT    HOSTED-CP SUPPORT
us-west-2    US West, Oregon    true                true

rosa list regions
ID                NAME                        MULTI-AZ SUPPORT    HOSTED-CP SUPPORT
us-east-1         US East, N. Virginia        true                false
us-east-2         US East, Ohio               true                false
us-west-1         US West, N. California      false               false
us-west-2         US West, Oregon             true                true

### ORG HostedCP NOT enabled
rosa list regions
ID                NAME                        MULTI-AZ SUPPORT
us-east-1         US East, N. Virginia        true
us-east-2         US East, Ohio               true
us-west-1         US West, N. California      false
us-west-2         US West, Oregon             true

### create account in not supported region
E: Region 'us-west-1' is not supported for Hosted Control Planes

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -1181,6 +1182,20 @@ func run(cmd *cobra.Command, _ []string) {
...
+		if isHostedCP && !interactive.Enabled() {
+			found := false
+			for i := range regionList {
+				if regionList[i] == region {
+					found = true
+					break
+				}
+			}
+			if !found {
+				r.Reporter.Errorf("Region '%s' is not supported for Hosted Control Planes", region)
```
Comment (pvasant) [inline cmd/create/cluster/cmd.go:1194]: s/Hosted Control planes/Hosted Cluster

File: cmd/list/region/cmd.go [hunk 2]
```diff
@@ -80,6 +89,16 @@ func run(cmd *cobra.Command, _ []string) {
...
 	}
 
+	hypershiftEnabled, err := r.OCMClient.IsCapabilityEnabled(ocm.HypershiftCapability)
+	if err != nil {
+		r.Reporter.Errorf("%v", err)
+		os.Exit(1)
+	}
+
+	if hypershiftEnabled {
+		regionsWithHostedCPSupport, _ = r.OCMClient.ListHostedCPSupportedRegion()
```
Comment (pvasant) [inline cmd/list/region/cmd.go:99]: why do we need to ignore the error

File: cmd/list/region/cmd.go [hunk 3]
```diff
@@ -110,15 +134,29 @@ func run(cmd *cobra.Command, _ []string) {
 
 	// Create the writer that will be used to print the tabulated results:
 	writer := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
-	fmt.Fprintf(writer, "ID\t\tNAME\t\tMULTI-AZ SUPPORT\n")
+	headerFormat := "ID\t\tNAME\t\tMULTI-AZ SUPPORT\n"
+	if hypershiftEnabled {
+		headerFormat = "ID\t\tNAME\t\tMULTI-AZ SUPPORT\t\tHOSTED-CP SUPPORT\n"
```
Comment (pvasant) [inline cmd/list/region/cmd.go:139]: @wgordon17 do we need to display Hosted-CP to the user or Hypershift Enabled??

File: cmd/create/cluster/cmd.go (hunk 1 above)
Comment (andreadecorte) [reply cmd/create/cluster/cmd.go]: fixed

File: cmd/list/region/cmd.go (hunk 2 above)
Comment (andreadecorte) [reply cmd/list/region/cmd.go]: fixed, thanks

File: cmd/list/region/cmd.go (hunk 3 above)
Comment (wgordon17) [reply cmd/list/region/cmd.go:143]: `HOSTED-CP` since `HyperShift` isn't meant for customer facing usage

**Pull request merged**